src/main/scala/org/dia/algorithms/mcc/GTGRunner.scala

```diff
@@ -324,18 +323,34 @@ class GTGRunner(val masterURL: String,
       minArea,
       nodeMinArea)
 
+    edgeListRDD.cache()
+    edgeListRDD.localCheckpoint()
+
     /**
      * Collect the edgeList and construct NodeMap
      */
     val MCCEdgeList = edgeListRDD.collect()
     val MCCNodeMap = createNodeMapFromEdgeList(MCCEdgeList, lat, lon)
 
+    val broadcastedNodeMap = sc.sparkContext.broadcast(MCCNodeMap)
+
     /**
      * Process the edge list. Collect and output edges and vertices
      */
     processEdges(MCCEdgeList, MCCNodeMap)
 
     /**
+     * Generate the netcdfs
+     */
+
+    val edgeTupleRDD = edgeListRDD.map({ x =>
+      (s"${x.srcNode.frameNum},${x.srcNode.cloudElemNum}",
+      s"${x.destNode.frameNum},${x.destNode.cloudElemNum}") })
+
+    val edgeTupleList = edgeTupleRDD.collect()
+    MCSUtils.get_all_node_data(edgeTupleList, broadcastedNodeMap.value, lat, lon, false)
```

Comment:
@sujen1412 
This is great. I'm guessing you're going to try to write it to HDFS?